datafusion/datasource-parquet/src/opener.rs

```diff
@@ -252,16 +257,20 @@ impl FileOpener for ParquetOpener {
             // This evaluates missing columns and inserts any necessary casts.
             let predicate = predicate
                 .map(|p| {
-                    PhysicalExprSchemaRewriter::new(
+                    let mut rewriter = PhysicalExprSchemaRewriter::new(
                         &physical_file_schema,
                         &logical_file_schema,
                     )
                     .with_partition_columns(
                         partition_fields.to_vec(),
                         file.partition_values,
-                    )
-                    .rewrite(p)
-                    .map_err(ArrowError::from)
+                    );
+                    if let Some(predicate_rewrite_hook) = predicate_rewrite_hook.as_ref()
+                    {
+                        rewriter = rewriter
+                            .with_rewrite_hook(Arc::clone(predicate_rewrite_hook));
+                    };
+                    rewriter.rewrite(p).map_err(ArrowError::from)
```

Comment:
I'm thinking maybe these should be decoupled and it just becomes two different rewrite passes.